sklearn/utils/testing.py

```diff
@@ -859,7 +860,8 @@ def check_docstring_parameters(func, doc=None, ignore=None):
     return incorrect
 
 
-def assert_run_python_script(source_code, timeout=60):
+def assert_run_python_script(source_code, timeout=60,
+                             ignore_init_warnings=True):
```

Comment:
> Doing nothing breaks several tests using assert_run_python_script because they expect no output but get a warning instead

Only in the newly created CI job, right? It seems to me that the other jobs should run just fine.

Reply:
Yes you're right.

Reply:
I find it unfortunate that we need a new parameter to this seemingly unrelated helper, just so that one of the CI doesn't complain.

Again, I don't have anything else to propose :/

But overall the changes proposed here are adding a lot of coupling in the code base, (that + the `sklearn._private_attr` thing).

I'm worried about the maintainability of all this.